simexpal/launch/common.py

```diff
@@ -453,9 +466,16 @@ def close(self):
 		# Consume any output that might be ready.
 		events = sel.select(timeout=1)
 		for (sk, mask) in events:
+			if sk.fileobj == signal_reader:
+				child.kill()
+				sys.exit(1)
```

Comment:
Instead of calling `exit(1)`, I think we should follow the normal code path to clean up (e.g., to write the status file), and then we should re-raise the signal or raise an exception.

Reply:
Done